Providing Windows scripts to install Zig and run the VOPR

SarahCoilAccount opened the pull request.

koekiebox commented on vopr.bat:
```diff
@@ -0,0 +1,25 @@
+
+:: Install Zig 0.8.0 if a zig folder does not already exist:
+if not exist zig\ (
+    call scripts/install_zig.bat 0.8.0
```
Change to; `scripts\install_zig.bat`

koekiebox commented on vopr.bat:
```diff
@@ -0,0 +1,25 @@
+    :: Build in fast ReleaseSafe mode if required, useful where you don't need debug logging:
+    if "%~2"=="-OReleaseSafe" (
+        echo Replaying seed %~1 in ReleaseSafe mode...
+        call zig run src/simulator.zig -OReleaseSafe -- %~1
```
Change to; `call zig\zig run src\simulator.zig -OReleaseSafe -- %~1`
We need to point to the `zig` instance that was just installed `(0.8.0)`.

## Otherwise
![2Capture](https://user-images.githubusercontent.com/1992108/134458975-51d4dfde-30eb-41f4-a7dd-5acec923444d.PNG)

koekiebox commented on vopr.bat:
```diff
@@ -0,0 +1,25 @@
+        call zig run src/simulator.zig -OReleaseSafe -- %~1
+    ) else (
+        echo Replaying seed %~1 in Debug mode with full debug logging enabled...
+        call zig run src/simulator.zig -ODebug -- %~1
```
Change to; `call zig\zig run src\simulator.zig -ODebug -- %~1`
We need to point to the `zig` instance that was just installed `(0.8.0)`

### Otherwise; 
![2Capture](https://user-images.githubusercontent.com/1992108/134458934-7f653463-efeb-4a14-aa17-a17b2ea29e04.PNG)
.

koekiebox commented on vopr.bat:
```diff
@@ -0,0 +1,25 @@
+        call zig run src/simulator.zig -ODebug -- %~1
+    )
+) else (
+    call zig build-exe src/simulator.zig -OReleaseSafe
```
Change to; `call zig\zig build-exe src\simulator.zig -OReleaseSafe`
We need to point to the `zig` instance that was just installed `(0.8.0)`

### Otherwise; 
![2Capture](https://user-images.githubusercontent.com/1992108/134458934-7f653463-efeb-4a14-aa17-a17b2ea29e04.PNG)
.

koekiebox reviewed (changes requested):
Had to make minor tweaks to point to the "just installed" instance of Zig.

# `install_zig.bat`
All good.

![1Capture](https://user-images.githubusercontent.com/1992108/134459167-45bf10b2-b286-4506-9aa2-18fbdbb07b8c.PNG)

# `vopr`

* Needs to point to `zig\zig`

## Error

![2Capture](https://user-images.githubusercontent.com/1992108/134459269-b87e4663-cc3c-4e37-ace8-d8dab7a96992.PNG)

koekiebox commented on scripts/install_zig.bat:
```diff
@@ -0,0 +1,59 @@
+@echo off
```
Please update the `/README.md` to reflect the support for Windows.

SarahCoilAccount commented:
I've addressed all these review changes. Thanks so much for the feedback. I wasn't getting errors for running zig instead of zig\zig because I have zig installed elsewhere already and had added it to my path! But I've fixed all that now and updated the README.

The PR is still incomplete now until I've also addressed Joran's comments:
- Add support for different versions of Zig, not just latest or 0.8.0.
- Add a regex check to ensure the ZIG_DIRECTORY variable is set to something expected before using it to remove the folders. 
- Implement better error handling.

koekiebox reviewed (approved):
Perfect @SarahCoilAccount. 🚀
